File: patterns/two-pointers/05.squares-of-sorted-array/main.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l = 0;
        int r = nums.size() - 1;
        vector<int> result = {};

        while (l <= r) {
            if (abs(nums[l]) > abs(nums[r])) {
                result.insert(result.begin(), nums[l] * nums[l]);
                l++;
            } else {
                result.insert(result.begin(), nums[r] * nums[r]);
                r--;
            }
        }

        return result;
    }
};

File: patterns/two-pointers/05.squares-of-sorted-array/main.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l = 0
        r = len(nums) - 1
        result = []

        while l <= r:
            if abs(nums[l]) > abs(nums[r]):
                result.insert(0, nums[l] ** 2)
                l += 1
            else:
                result.insert(0, nums[r] ** 2)
                r -= 1

        return result
